### PR TITLE
Update customisation options to match the vscode extension

### PR DESCRIPTION
**Pull request by alanz**
In particular, add the ability to turn on the new cabal plugin and completion for it.

### CODE REVIEW

**Comment by alanz** [comment]
I am not able to get this working locally, so probably not a good idea to merge yet.

**Comment by michaelpj** [comment]
BTW you want to use the schema produced by `haskell-language-server vscode-extension-schema`, not the one from `vscode-haskell`. They add a bunch of options that are actually specific to the vscode extension rather than HLS.

**Comment by alanz** [comment]
And it struck me I need to ensure it actually gets used too.

Where is the best place to document this update process?

**Comment by michaelpj** [comment]
> And it struck me I need to ensure it actually gets used too.

I believe `lsp-defcustom` handles this for us.

> Where is the best place to document this update process?

Start a CONTRIBUTING.md?

**Comment by alanz** [comment]
Take two.  Turns out if you run the update script with `lsp-haskell` already active, it only presents settings for the items not already present.

**Comment by alanz** [comment]
I find it now works, in that it starts up correctly, and the config settings are sent.  But for me completions do not work.

```json
[Trace - 12:38:28 pm] Sending notification 'workspace/didChangeConfiguration'.
Params: {
  "settings": {
    "haskell": {
      "formattingProvider": "fourmolu",
      "checkProject": true,
      "maxCompletions": 40,
      "plugin": {
        "importLens": {
          "codeActionsOn": false,
          "codeLensOn": false
        },
        "hlint": {
          "codeActionsOn": true,
          "diagnosticsOn": true
        },
        "eval": {
          "globalOn": true
        },
        "moduleName": {
          "globalOn": true
        },
        "splice": {
          "globalOn": true
        },
        "haddockComments": {
          "globalOn": true
        },
        "class": {
          "globalOn": true
        },
        "retrie": {
          "globalOn": true
        },
        "stan": {
          "globalOn": true
        },
        "tactics": {
          "globalOn": true,
          "config": {
            "auto_gas": 4,
            "max_use_ctor_actions": 5,
            "timeout_duration": 2,
            "proofstate_styling": true
          }
        },
        "pragmas": {
          "codeActionsOn": true,
          "completionsOn": true
        },
        "ghcide-completions": {
          "config": {
            "autoExtendOn": true,
            "snippetsOn": true
          }
        },
        "ghcide-type-lenses": {
          "globalOn": false,
          "config": {
            "mode": "diagnostics"
          }
        },
        "refineImports": {
          "globalOn": false
        },
        "cabal": {
          "codeActionsOn": true,
          "completionOn": true
        },
        "overloaded-record-dot": {
          "globalOn": true
        },
        "pragmas-completion": {
          "globalOn": true
        },
        "pragmas-disable": {
          "globalOn": true
        },
        "pragmas-suggest": {
          "globalOn": true
        }
      }
    }
  }
}

```

File: CONTIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Contributing to lsp-haskell
```
Comment (michaelpj) [inline CONTIBUTING.md:1]: Great stuff, thanks

File: lsp-haskell.el [hunk 2]
```diff
@@ -47,13 +47,15 @@
 
 (defcustom-lsp lsp-haskell-formatting-provider
```
Comment (michaelpj) [inline lsp-haskell.el:48]: could migrate these to `lsp-defcustom` while we're at it. Not sure if it's just a rename or what

**Review by michaelpj — APPROVED** [review]

**Comment by michaelpj** [comment]
We should debug what's going on with cabal files. Is the mode definitely activating? Do you see `lsp-mode` sending messages to HLS when you're in a cabal file?

**Pull request merged**